packages/compass-query-bar/src/utils/query-storage.ts

```diff
@@ -2,41 +2,66 @@ import { UUID, EJSON } from 'bson';
 import { orderBy } from 'lodash';
 import { type BaseQuery } from '../constants/query-properties';
 import { UserData } from '@mongodb-js/compass-user-data';
+import { z } from 'zod';
 
 // We do not save maxTimeMS
-export type RecentQuery = Omit<BaseQuery, 'maxTimeMS'> & {
-  _id: string;
-  _lastExecuted: Date;
-  _ns: string;
-  _host?: string;
-};
+const BaseQuerySchema: z.Schema<Omit<BaseQuery, 'maxTimeMS'>> = z.object({
+  filter: z.any().optional(),
+  project: z.any().optional(),
+  collation: z.any().optional(),
+  sort: z.any().optional(),
+  skip: z.number().optional(),
+  limit: z.number().optional(),
+});
 
-export type FavoriteQuery = RecentQuery & {
-  _name: string;
-  _dateModified: Date;
-  _dateSaved: Date;
-};
+const RecentQuerySchema = BaseQuerySchema.and(
+  z.object({
+    _id: z.string().uuid(),
+    _lastExecuted: z
+      .union([z.coerce.date(), z.number()])
+      .transform((x) => new Date(x)),
+    _ns: z.string(),
+    _host: z.string().optional(),
+  })
+);
+
+const FavoriteQuerySchema = RecentQuerySchema.and(
+  z.object({
+    _name: z.string().nonempty(),
+    _dateModified: z
+      .union([z.coerce.date(), z.number()])
+      .optional()
+      .transform((x) => (x ? new Date(x) : x)),
```

Comment:
```suggestion
      .transform((x) => (x !== undefined ? new Date(x) : x)),
```

otherwise this fails to bootstrap for me because `FavoriteQuery['_dateModified']` could be `0`, type-wise